modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/loader/XmlExtensionLoaderDelegate.java

```diff
@@ -249,6 +243,12 @@ private void loadModuleExtension(ExtensionDeclarer declarer, ComponentModel modu
     }
   }
 
+  private XmlDslModel getXmlDslModel(ComponentModel moduleModel, String name, String version) {
```

Comment:
it's similar, but not the same. 
The @Xml annotation might not be present, that's why it has a different way of accessing the valuea